pkg/entities/image/actions.go

```diff
@@ -111,35 +112,51 @@ func (action *buildImage) imageExists() (bool, error) {
 	}
 
 	imageDestroyDir, _ := filepath.Split(action.img.configPath)
+	logger := log.WithFields(log.Fields{
+		"dir":   imageDestroyDir,
+		"image": action.img,
+	})
 
 	tfLogPrefix, err := action.img.makeToolLogPrefix("terraform")
 	if err != nil {
-		log.WithFields(log.Fields{
-			"image": action.img,
-		}).Warnf("Image.imageExists: cannot make logfile name: %s", err)
+		logger.Warnf("Image.imageExists: cannot make logfile name: %s", err)
 	}
 
 	action.stage.Set(":checking existence")
 	defer action.stage.Reset()
 
+	if _, err := action_pkg.RunLoggedCmdDir(tfLogPrefix, imageDestroyDir, provider.Terraform(),
+		"refresh", "-state-out=checked.tfstate", "-backup=-"); err != nil {
+		if exited, ok := err.(*exec.ExitError); ok {
+			if exited.ExitCode() != -1 {
+				logger.Info("Image.imageExists: refreshed failed; assume image does not exist")
+			}
+			return false, nil
+		}
+	}
+
+	var buffer0 bytes.Buffer
+	if logname, err := action_pkg.RunLoggedCmdDirOutput(tfLogPrefix, imageDestroyDir, &buffer0, provider.Terraform(),
+		"output", "-state=checked.tfstate", "id"); err != nil {
+		logger.Errorf("Image.imageExists: terraform output failed: %s", err)
```

Comment:
```suggestion
		logger.Errorf("Image.imageExists: 'terraform output' failed: %s", err)
```
to highlight what exactly failed